### ISSUE TITLE
Duplicate msgid with singular and plural form

### ISSUE THREAD

**Issue by saveman71** [elixir-gettext/gettext#366]
Not sure if it's a bug / how other implementations tackle that, but `msgfmt` will fail for duplicate `msgid`s:

```
❯ msgfmt priv/gettext/default.pot
priv/gettext/default.pot:10: duplicate message definition...
priv/gettext/default.pot:5: ...this is the location of the first definition
msgfmt: found 1 fatal error
```

Which is expected for cases like:

```
#, elixir-autogen, elixir-format
msgid "Singular."
msgstr ""

#, elixir-autogen, elixir-format
msgid "Singular."
msgstr ""
```

However, it also fails for cases like:

```
#: lib/gettext_mwe.ex:20
#, elixir-autogen, elixir-format
msgid "Singular."
msgid_plural "Plural."
msgstr[0] ""
msgstr[1] ""

#: lib/gettext_mwe.ex:18
#, elixir-autogen, elixir-format
msgid "Singular."
msgstr ""
```

Generated by this code:

```
defmodule GettextMwe do
  import GettextMwe.Gettext

  @moduledoc false

  def singular do
    gettext("Singular.")
  end

  def plural(count) do
    ngettext("Singular.", "Plural.", count)
  end
end
```

I believe things should work file if the singular form was dropped from the `po`/`pot` file.

Here's the repro repo: https://github.com/saveman71/elixir-gettext-duplicate-keys

**Comment by whatyouhide** [comment]
We can't merge the translations yeah. I mean, the Elixir Gettext behavior seems fine to me because the singular and plural translations are treated completely differently, so they can coexist with the same `msgid`.

Is `msgfmt` the only thing that fails on this in the GNU Gettext toolset? Also, do you know what GNU Gettext does when extracting in situations when this would happen? For example, I don't know how PHP's [ngettext](https://www.php.net/manual/en/function.ngettext.php) or other implementations deal with this.

**Comment by maennchen** [comment]
@whatyouhide PHP reads the `msgid` of a plural translation as well:

https://github.com/php/php-src/blob/9e15910889cfa1f14169ed9babf3bf54426cca1c/ext/gettext/tests/gettext_dgettext.phpt#L23

```php
var_dump(gettext('item'));
[...]
string(7) "Produkt"
```

https://github.com/php/php-src/blob/afffcaeb4d994be50c83ac0204b4d4be8af560d9/ext/gettext/tests/locale/en_US.UTF-8/LC_MESSAGES/dngettextTest.po#L1

```po
msgid "item"
msgid_plural "items"
msgstr[0] "Produkt"
msgstr[1] "Produkte"
```

**Comment by maennchen** [comment]
Judging by this paragraph, I think that is the expected way to do things:

https://www.gnu.org/software/gettext/manual/gettext.html#Invoking-the-msguniq-Program

> The msguniq program unifies duplicate translations in a translation catalog. It finds duplicate translations of the same message ID. Such duplicates are invalid input for other programs like msgfmt, msgmerge or msgcat. By default, duplicates are merged together. When using the ‘--repeated’ option, only duplicates are output, and all other messages are discarded. Comments and extracted comments will be cumulated, except that if ‘--use-first’ is specified, they will be taken from the first translation. File positions will be cumulated. When using the ‘--unique’ option, duplicates are discarded.

So I guess we should merge singular and plural into one plural message and use the `msgid` of plural messages for singular calls.

**Comment by whatyouhide** [comment]
Yeah I think that's a good idea. I will be out all of next week so it will be a while before I'll be able to do anything about this 🙃

**Comment by saveman71** [comment]
Sorry I was out of office and not great at using GitHub's notification center :sweat_smile: Thanks for looking into this!
